create User model, migration files and tests

AshkenYariv opened the pull request:
Creating a user model, a static method for creating users.
Made migration files for user.
Wrote tests for user creation.
Deleted joboard application from the first demo since
it is not relevant anymore.

yuvalk commented:
plz separate the removal of the board content from the adding of the user model.
(so you should have 2 commits, one that remove stuff and another one that add new stuff)

yuvalk commented:
also, try to be as extensive as possible with tests, cover all the options. try to think about edge cases

AshkenYariv committed:
Deletion of jobboard app

Deletion of joboard application from the first demo since
it is not relevant anymore.

AshkenYariv committed:
Create user model, migration files and tests

Creating a user model, a static method for creating users.
Made migration files for user.
Wrote tests for user creation.

shayakrach commented:
@yuvalk  We split up the PR.

AshkenYariv committed:
Creation jobs model, migration files and tests

Creation of Jobs model.
Migrating the Jobs model.
Adding tests to model and functions.

AshkenYariv committed:
Register users and jobs model to admin

Registering users and jobs model to admin.

ormergi reviewed (commented):
Regarding Commit "Deletion of jobboard app", it can be extracted to its own PR, as its not directly related to adding models.
Also keeping PR's small reduce code-review cycles and keep your reviewers focused.

ormergi commented on side_hustle/settings.py:
```diff
@@ -37,7 +37,7 @@
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
-    'jobboard.apps.JobboardConfig',
```
This changes is related to removing jobboard app commit, isnt it?

shayakrach replied:
Right.

ormergi replied:
Please move it to the first commit

ormergi commented on side_hustle/urls.py:
```diff
@@ -15,9 +15,7 @@
 """
 from django.contrib import admin
 from django.urls import path
-from jobboard import views
 
 urlpatterns = [
     path('admin/', admin.site.urls),
-    path('', views.board, name='board')
```
Same goes for these, they are related to the commit before this one, right?

ormergi reviewed (commented):
Regarding "Create user model, migration files and tests" commit
The commit title should not describe adding tests or migration files, these are implementation details that could be understand by the actual changes this commit beings.
It can be rephrased to something like _'Add 'User' data model"_.
Same goes for the the commit message body, the "what" is described byt the actual changes this commit brings (add/remove/change files).

ormergi reviewed (commented):
Regarding "Creation jobs model, migration files and tests" commit, see my comment about the previous commit https://github.com/redhat-beyond/Side-Hustle/pull/21#pullrequestreview-817147622.

shayakrach commented:
@ormergi 
We tried to keep the first commit separate and continue with another branch not dependent on it, but the jobborad app caused some errors. What should we do in that situation? And can we fix the current situation?

ormergi reviewed (commented):
Regarding "Register users and jobs model to admin" commit,
correct me if I am wrong, but it looks like this commit has more to it other than what the message describes.
Those changes should be moved to new commit(s) when needed

ormergi commented:
> @ormergi We tried to keep the first commit separate and continue with another branch not dependent on it, but the jobborad app caused some errors. What should we do in that situation? And can we fix the current situation?

There are few changes on the **second** commit that related to the **first** one, maybe its related.
Move the related changes to the first commit and try again.

shayakrach commented:
@ormergi We understand **what** is the problem but we need help with the git technicalities.

AshkenYariv commented:
Hello @ormergi these comit have been up on GItHub since sunday and we made correction according to what Edward and Yuval told us, these are too many changes that should have came up at an earlier time. What do you suggest we do taking into account that tomorrow we need to show Demo#2?

shayakrach commented:
@ormergi We opened new PR #24.

EdDev commented:
Please rebase over the updated master, this PR has conflicts now.

AshkenYariv commented:
This PR was split up as asked, the reviewing should start at PR create user model

ormergi commented:
@AshkenYariv its good that you explain why you close the PR, it will be awesome if you also paste the like for the new one.
Also, my code review comments may help you in following PR's as well.

EDIT: Cant find the following PR, if its not published yet please mention it.

shayakrach commented:
Hi @ormergi, 
We forgot to paste it.
The following PR is #25.
